rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -0,0 +1,271 @@
+          case None => Full("ok")
+        }
+      } yield {
+        file
+
+      }
+    }
+  }
+
+  def writeMetadata(technique : Technique, methods: Map[BundleName, GenericMethod]) : Box[Unit] = {
+    import net.liftweb.util.Helpers.tryo
+    tryo {
+      val content = commonMetadata(technique, methods)
+      val path = s"/var/rudder/configuration-repository/${metadataPath(technique)}"
+      val filePath = Paths.get(path)
+      if (!Files.exists(filePath))
+        Files.createFile(filePath)
+      scala.xml.XML.save(path, content)
+    }
+  }
+}
+
+trait AgentSpecificWriter {
+
+  def writeTechniqueFile(technique : Technique, methods : Map[BundleName, GenericMethod] ) : Box[Option[String]]
+
+  def techniqueMetadata(technique : Technique, methods : Map[BundleName, GenericMethod] ) : NodeSeq
+}
+
+class ClassicTechniqueWriter extends AgentSpecificWriter {
+
+  def writeTechniqueFile(technique : Technique, methods : Map[BundleName, GenericMethod]) = Full(None)
+  def techniqueMetadata(technique : Technique, methods : Map[BundleName, GenericMethod]) = {
+    val completeReporting = technique.methodCalls.exists(m => m.condition != "any" || m.condition != "cfengine-community" )
+    <AGENT type="cfengine-community,cfengine-nova">
+      <BUNDLES>
+        <NAME>{technique.bundleName.value}</NAME>
+        {if (completeReporting) <NAME>{technique.bundleName.value}_rudder_reporting</NAME>}
+      </BUNDLES>
+      { if (completeReporting)
+        <TMLS>
+          <TML name="rudder_reporting"/>
+        </TMLS>
+      }
+      <FILES>
+        <FILE name={s"RUDDER_CONFIGURATION_REPOSITORY/ncf/50_techniques/${technique.bundleName.value}/${technique.bundleName.value}.cf"}>
+          <INCLUDED>true</INCLUDED>
+        </FILE>
+      </FILES>
+    </AGENT>
+  }
+
+}
+
+class DSCTechniqueWriter extends AgentSpecificWriter{
+
+  val genericParams = "-reportId $reportId -techniqueName $techniqueName -auditOnly:$auditOnly"
+
+  def writeTechniqueFile(technique : Technique, methods : Map[BundleName, GenericMethod] ) = {
+
+  def writeCall(method : GenericMethod, call: MethodCall) = {
+    val methodParams = (for {
+      (id, arg) <- call.parameters
+    } yield {
+      s"""-${id.toDsc} "${arg.replaceAll("\"", "`\"")}""""
+    }).mkString(" ")
+    s"""$$local_classes = Merge-ClassContext $$local_classes $$(${method.id.toDsc} ${methodParams} ${genericParams}).get_item("classes")"""
+  }
+  def na_report(method : GenericMethod, call: MethodCall) =
+    s"""_rudder_common_report_na -componentName "${method.name}" -componentKey "${call.parameters.get(method.classParameter).map(_.replaceAll("\"", "`\""))}" -message "Not applicable" ${genericParams}"""
+
+  def methodCall(call : MethodCall) = {
+    methods.get(call.methodId) match {
+      case Some(method) if method.agentSupport.contains(AgentType.Dsc) =>
+        call.condition match {
+          case "any" =>
+            writeCall(method, call)
+          case _ =>
+
+         s"""
```

Comment:
would you mind use "|" + .stripMargin ? I find it quite hard to read here. Moreover, you are adding a bunch of spaces in the last line, I'm almost sure it will cause troubles when testing.